### PR TITLE
Enhancement/allow multiple users

### PR DESCRIPTION
**Pull request by ghost**
### What changes were proposed in this pull request?

Allow the configuration of multiple users in the user details configuration.

### How is this patch documented?

README.adoc, javadoc and configuration file example 

### How was this patch tested?

Postman and existing tests

#### Depends On

https://github.com/SMARTRACTECHNOLOGY/smartcosmos-cluster-config/pull/23

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -41,14 +41,14 @@ Not to be confused with the credentials the **Auth Server** used to authenticate
 
 == Configuration
 
-The user information and authorities returned by the service can be configured:
+The user information and authorities returned by the service are configured in the service config.  Multiple users can eb configured as shown in the following example.
```
Comment (andreassiegel) [inline README.adoc]: Typo: `eb` instead of `be`

File: README.adoc [hunk 2]
```diff
@@ -41,14 +41,14 @@ Not to be confused with the credentials the **Auth Server** used to authenticate
...
 
 [source,yaml]
 ----
 smartcosmos:
   security:
-    user:
-      name: user
+    users[user]:
```
Comment (andreassiegel) [inline README.adoc:50]: Isn't it possible to do something like this?

```
users:
    - username: user
      password: password
      ...
    - username: user2
      password: password2
      ...
```

In Ansible (which also uses YML) this syntax works perfectly.

File: src/main/java/net/smartcosmos/cluster/userdetails/DevelopmentUserDetailsService.java [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
-import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Import;
 
-import net.smartcosmos.cluster.userdetails.config.UserAuthenticationProperties;
+import net.smartcosmos.cluster.userdetails.config.UserDetailsDevelopmentConfiguration;
 
 @SpringBootApplication
-@EnableConfigurationProperties({ UserAuthenticationProperties.class })
+@EnableAutoConfiguration
+@Import(UserDetailsDevelopmentConfiguration.class)
```
Comment (andreassiegel) [inline src/main/java/net/smartcosmos/cluster/userdetails/DevelopmentUserDetailsService.java:12]: You probably don't need this if the `@ComponentScan` can locate your `@Configuration` class.

**Review by andreassiegel — COMMENTED** [review]
Looks good in general, but there are a few things that probably can still be improved.

File: README.adoc (hunk 2 above)
Comment (ghost) [reply README.adoc:50]: yes but that is ugly to me and is not a map, I want a map.  I want to lookup by key not search a list with contains

File: src/main/java/net/smartcosmos/cluster/userdetails/DevelopmentUserDetailsService.java (hunk 3 above)
Comment (ghost) [reply src/main/java/net/smartcosmos/cluster/userdetails/DevelopmentUserDetailsService.java:12]: I've been having problems with autowiring.  I am trying to get tests going that do not require loading the entire works and all the web mvn components.  It's still not correct

File: README.adoc (hunk 2 above)
Comment (andreassiegel) [reply README.adoc:50]: You're right, that makes perfect sense!

**Review by andreassiegel — APPROVED** [review]

**Pull request merged**